### PR TITLE
Fix signature on query method, and add test with null

### PR DESCRIPTION
**Pull request by nielsenko**

### CODE REVIEW

File: test/results_test.dart [hunk 1]
```diff
@@ -315,6 +315,20 @@ Future<void> main([List<String>? args]) async {
     expect(() => teams.length, throws<RealmException>("Access to invalidated Results objects"));
   });
 
+  test('Results - query with nil', () {
```
Comment (blagoev) [inline test/results_test.dart:318]:
```suggestion
  test('Results - query with null', () {
```

**Review by blagoev — APPROVED** [review]
